core/tests/protractor_utils/forms.js

```diff
@@ -44,217 +48,217 @@ var GraphEditor = function(graphInputContainer) {
       '.protractor-test-graph-vertex-' + index));
   };
 
-  var createVertex = function(xOffset, yOffset) {
+  var createVertex = async function(xOffset, yOffset) {
     var addNodeButton = graphInputContainer.element(
       by.css('.protractor-test-Add-Node-button'));
-    addNodeButton.click();
+    await addNodeButton.click();
     // Offsetting from the graph container.
-    browser.actions()
+    await browser.actions()
       .mouseMove(graphInputContainer, {x: xOffset, y: yOffset})
       .click()
       .perform();
   };
 
-  var createEdge = function(vertexIndex1, vertexIndex2) {
+  var createEdge = async function(vertexIndex1, vertexIndex2) {
     var addEdgeButton = graphInputContainer.element(
       by.css('.protractor-test-Add-Edge-button'));
-    addEdgeButton.click();
-    browser.actions()
+    await addEdgeButton.click();
+    await browser.actions()
       .dragAndDrop(vertexElement(vertexIndex1), vertexElement(vertexIndex2))
       .perform();
   };
   return {
-    setValue: function(graphDict) {
+    setValue: async function(graphDict) {
       var nodeCoordinatesList = graphDict.vertices;
       var edgesList = graphDict.edges;
       if (nodeCoordinatesList) {
         expect(nodeCoordinatesList.length).toBeGreaterThan(0);
         // Assume x-coord is at index 0.
-        nodeCoordinatesList.forEach(function(coordinateElement) {
-          createVertex(coordinateElement[0], coordinateElement[1]);
-        });
+        for (coordinateElement of nodeCoordinatesList) {
+          await createVertex(coordinateElement[0], coordinateElement[1]);
+        }
       }
       if (edgesList) {
-        edgesList.forEach(function(edgeElement) {
-          createEdge(edgeElement[0], edgeElement[1]);
-        });
+        for (edgeElement of edgesList) {
+          await createEdge(edgeElement[0], edgeElement[1]);
+        }
       }
     },
-    clearDefaultGraph: function() {
+    clearDefaultGraph: async function() {
       var deleteButton = graphInputContainer.element(
         by.css('.protractor-test-Delete-button'));
-      deleteButton.click();
+      await deleteButton.click();
       // Sample graph comes with 3 vertices.
       for (var i = 2; i >= 0; i--) {
-        vertexElement(i).click();
+        await vertexElement(i).click();
       }
     },
-    expectCurrentGraphToBe: function(graphDict) {
+    expectCurrentGraphToBe: async function(graphDict) {
       var nodeCoordinatesList = graphDict.vertices;
       var edgesList = graphDict.edges;
       if (nodeCoordinatesList) {
         // Expecting total no. of vertices on the graph matches with the given
         // dict's vertices.
-        nodeCoordinatesList.forEach(function(node, index) {
-          expect(vertexElement(index).isDisplayed()).toBe(true);
-        });
+        for (var i = 0; i < nodeCoordinatesList.count(); i++) {
```

Comment:
I think it should be `nodeCoordinatesList.length`. This is a JavaScript Array and it does not have a `count()` method

Reply:
Appears fixed in current diff